packages/odyssey-react-mui/src/labs/FileUploadPreview.tsx

```diff
@@ -0,0 +1,158 @@
+  }),
+
+  ({ isDisabled }) => ({
+    pointerEvents: isDisabled ? "none" : "auto",
+  }),
+);
+
+const UploadedFileContainer = styled.div<{ odysseyDesignTokens: DesignTokens }>(
+  {
+    display: "flex",
+    justifyContent: "space-between",
+    alignItems: "center",
+
+    button: {
+      transform: "scale(0)",
+    },
+
+    "&:hover, &:focus-within, &:focus": {
+      button: {
+        transform: "scale(1)",
+      },
+    },
+  },
+  ({ odysseyDesignTokens }) => ({
+    padding: `${odysseyDesignTokens.Spacing1} ${odysseyDesignTokens.Spacing2}`,
+    borderRadius: odysseyDesignTokens.BorderRadiusMain,
+    transition: `background-color ${odysseyDesignTokens.TransitionTimingMain}`,
+
+    "&:hover, &:focus-within": {
+      backgroundColor: odysseyDesignTokens.HueNeutral100,
+    },
+
+    "&:focus": {
+      borderColor: odysseyDesignTokens.FocusOutlineColorPrimary,
+      boxShadow: `0 0 0 2px ${odysseyDesignTokens.FocusOutlineColorPrimary}`,
+      outline: `${odysseyDesignTokens.FocusOutlineWidthMain} ${odysseyDesignTokens.FocusOutlineStyle} transparent`,
+    },
+  }),
+);
+
+type UploadedFileProps = {
+  name: string;
+  onFileRemove?: (name: string) => void;
+};
+
+const UploadedFile = ({ name, onFileRemove }: UploadedFileProps) => {
+  const { t } = useTranslation();
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const deleteHandler = useCallback(() => {
+    onFileRemove?.(name);
+  }, [onFileRemove, name]);
+
+  const renderDeleteButton = useCallback(
+    (muiProps: MuiPropsContextType) => {
+      return (
+        <IconButton
+          {...muiProps}
+          aria-label={t("fileupload.removefile.text")}
+          onClick={deleteHandler}
+          size="small"
+        >
+          <DeleteIcon />
+        </IconButton>
+      );
+    },
+    [deleteHandler],
+  );
+
+  return (
+    <UploadedFileContainer
+      // tabindex added to make div focusable
+      tabIndex={0}
+      odysseyDesignTokens={odysseyDesignTokens}
+    >
+      {name}
+      <Tooltip
+        ariaType="description"
+        placement="top"
+        text={t("fileupload.removefile.text")}
+      >
+        <MuiPropsContext.Consumer>
+          {renderDeleteButton}
+        </MuiPropsContext.Consumer>
+      </Tooltip>
+    </UploadedFileContainer>
+  );
+};
+
+type FileUploadPreviewProps = {
+  fileNames: string[];
+  isDisabled: FileUploadProps["isDisabled"];
+  onFileRemove?: (name: string) => void;
+};
+
+const FileUploadPreview = ({
+  fileNames,
+  isDisabled,
+  onFileRemove,
+}: FileUploadPreviewProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  return (
+    <PreviewContainer
+      isDisabled={isDisabled}
+      odysseyDesignTokens={odysseyDesignTokens}
+    >
+      {fileNames?.map((name: string, index: number) => (
+        <UploadedFile
+          key={`${index}-${name}`}
+          onFileRemove={onFileRemove}
+          name={name}
+        />
+      ))}
```

Comment:
Since we know the types of `fileNames`, we don't need to re-type them in the `.map`:
```suggestion
      {fileNames?.map((name, index) => (
        <UploadedFile
          key={`${index}-${name}`}
          onFileRemove={onFileRemove}
          name={name}
        />
      ))}
```